### PR TITLE
add additonal trusted CA option to eks-operator

### PR DESCRIPTION
**Pull request by bentastic27**
https://github.com/rancher/rancher/issues/31846

### CODE REVIEW

**Review by StrongMonkey — DISMISSED** [review]
LGTM, But you will also need to make changes in rancher to pass down values.yaml option. For example, https://github.com/rancher/rancher/blob/b44c2232a71b5004f0de66833ed111540f11c6c7/pkg/controllers/management/clusteroperator/utils.go#L189

File: charts/rancher-eks-operator/1.0.800/templates/deployment.yaml [hunk 1]
```diff
@@ -25,3 +25,17 @@ spec:
           value: {{ .Values.httpsProxy }}
         - name: NO_PROXY
           value: {{ .Values.noProxy }}
+{{- if .Values.additionalTrustedCAs }}
+        volumeMounts:
+        - mountPath: /etc/ssl/certs/ca-additional.pem
+          name: tls-ca-additional-volume
+          subPath: ca-additional.pem
```
Comment (thedadams) [inline charts/rancher-eks-operator/1.0.800/templates/deployment.yaml]: Should the `subPath` from the secret be configurable in the `values.yaml`?

File: charts/rancher-eks-operator/1.0.800/templates/deployment.yaml [hunk 2]
```diff
@@ -25,3 +25,17 @@ spec:
...
+          readOnly: true
+{{- end }}
+{{- if .Values.additionalTrustedCAs }}
+      volumes:
+      - name: tls-ca-additional-volume
+        secret:
+          defaultMode: 0400
+          secretName: tls-ca-additional
```
Comment (thedadams) [inline charts/rancher-eks-operator/1.0.800/templates/deployment.yaml]: Should the `secretName` be configurable in the `values.yaml`?

**Review by rmweir — DISMISSED** [review]
LGTM, but the changes to rancher/rancher mentioned by @StrongMonkey need to be made before this can move to test.

**Review by thedadams — CHANGES_REQUESTED** [review]
Since version 1.0.800 has been released, I believe that the 1.0.800 folder needs to be copied to 1.0.801 and these changes made there (as well as the corresponding version bumps).

**Review by thedadams — DISMISSED** [review]

**Review by rmweir — DISMISSED** [review]
As long as the min version is intended, LGTM.

File: charts/rancher-eks-operator/1.0.801/Chart.yaml [hunk 3]
```diff
@@ -4,5 +4,5 @@ description: A Helm chart for provisioning EKS clusters
 home: https://github.com/rancher/eks-operator
 sources:
   - "https://github.com/rancher/eks-operator"
-version: 1.0.800
+version: 1.0.801
```
Comment (cbron) [inline charts/rancher-eks-operator/1.0.801/Chart.yaml:7]: Are we going to tag a 1.0.9 for 2.5.10 ?

File: charts/rancher-eks-operator/1.0.801/questions.yaml [hunk 4]
```diff
@@ -0,0 +1 @@
+rancher_min_version: 2.5.6-alpha1
```
Comment (cbron) [inline charts/rancher-eks-operator/1.0.801/questions.yaml]: This should be 2.5.10 and last chart should max at 2.5.9

File: charts/rancher-eks-operator/1.0.801/Chart.yaml (hunk 3 above)
Comment (thedadams) [reply charts/rancher-eks-operator/1.0.801/Chart.yaml:7]: No, I don't believe so.

File: charts/rancher-eks-operator/1.0.801/questions.yaml (hunk 4 above)
Comment (thedadams) [reply charts/rancher-eks-operator/1.0.801/questions.yaml]: This is done.

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Review by cbron — APPROVED** [review]

**Pull request merged**